common/lib/api-client/middleware/cache-key.js

```diff
@@ -0,0 +1,30 @@
+const { get } = require('lodash')
+
+const models = require('../models')
+
+function cacheKeyMiddleware({ apiVersion = 2 } = {}) {
+  return {
+    name: 'cache-key',
+    req: function req(payload) {
+      const { req } = payload
+      const cacheModel = get(models, `${req.model}.options.cache`)
+
+      const tryCache = cacheModel && req.params.cache !== false
+
+      if (tryCache) {
+        const pathname = new URL(req.url).pathname
+
+        const searchString = new URLSearchParams(req.params).toString()
+        const key = `cache:v${apiVersion}:${req.method}.${pathname}${
+          searchString ? `?${searchString}` : ''
+        }`
+
+        payload.cacheKey = key
```

Comment:
Would this be better placed on the request object seeing as it relates to the request - `payload.req.cacheKey`?

Reply:
I don't see that it makes much odds but I think it's better on the payload

Reply:
As per https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/pull/801#discussion_r492632004

Should it throw if not passed a defined value?

Reply:
It just feels more closely aligned to the request than the payload in this instance. Like `req.url`, `req.params`, `req.model`....the cache key is more tied to the request in this instance than the payload.

Reply:
`req` is what get passed to `axios`. The `cacheKey` has nothing to do with it. We are proper 🚲  🏠 ing here